src/App.tsx

```diff
@@ -1,61 +1,135 @@
+import { ChangeEvent, useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const getUserById = (userId: number): User | null => (
+  usersFromServer.find(user => user.id === userId) || null
+);
+
+const getUserByName = (name: string): User | null => (
+  usersFromServer.find(user => user.name === name) || null
+);
+
+const refactoredTodos: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(refactoredTodos);
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState('');
+  const [titleValid, setTitleValid] = useState(true);
+  const [userValid, setUserValid] = useState(true);
+
+  const handleTitleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleValid(Boolean(event.target.value));
+  };
```

Comment:
I can push todo with spaces in title

![image](https://github.com/mate-academy/react_add-todo-form/assets/84296041/bf611a43-f49d-4fce-90f6-e10a9c3902e0)